### PR TITLE
Feature enum my inject

### PR DESCRIPTION
**Pull request by rmauritsson**

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            new_array
+        else
+            self.my_each do |x|
+                new_array << proc.call(x)
+            end
+            new_array
+        end
+    end
+
+    def my_inject
```
Comment (Ispirett) [inline main.rb]: my, inject should be able to take an argument.

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        self.my_each { |item| eturnr false if yield(item) == false }
+        true
+    end
+
+    def my_any?
+        self.my_each { |item| return true if yield(item) }
+        false
+    end
+
+    def my_none?
```
Comment (Ispirett) [inline main.rb]: **my_count**  should be able to take an argument.

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        self.my_each { |item| result.push(item) if yield(item) }
+        result
+    end
+
+    def my_all?
+        self.my_each { |item| eturnr false if yield(item) == false }
+        true
+    end
+
+    def my_any?
```
Comment (Ispirett) [inline main.rb]: **my_any?** should be able to take an argument.
your method should also have a guard case that handles and error when **no block is given**

File: main.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        end
+    end
+
+    def my_select
+        result = []
+        self.my_each { |item| result.push(item) if yield(item) }
+        result
+    end
+
+    def my_all?
```
Comment (Ispirett) [inline main.rb]: **my_all?** should be able to take an argument.
**your method should also have a guard case that handles and error when no block is given**
 undefined method `eturnr'

File: main.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    def my_each_with_index
+        i = 0
+        while i < self.size
+            yield(self[i], i)
+            i += 1
+            self
+        end
+    end
+
+    def my_select
```
Comment (Ispirett) [inline main.rb]: your method should also have a guard case that handles and error when no block is given

File: main.rb [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    def my_each
+        i = 0
+        while i < self.size
+            yield (self[i])
+            i += 1
+        end
+        self
+    end
+
+    def my_each_with_index
```
Comment (Ispirett) [inline main.rb]: your method should also have a guard case that handles and error when no block is given

File: main.rb [hunk 7]
```diff
@@ -0,0 +1,90 @@
+module Enumerable
+    def my_each
```
Comment (Ispirett) [inline main.rb]: your method should also have a guard case that handles and error when no block is given

File: main.rb (hunk 1 above)
Comment (rmauritsson) [reply main.rb]: Fixed

File: main.rb (hunk 2 above)
Comment (rmauritsson) [reply main.rb]: Done

File: main.rb (hunk 3 above)
Comment (rmauritsson) [reply main.rb]: Done

**Comment by rmauritsson** [comment]
Completed all the changes suggested

**Comment by rmauritsson** [comment]
Modified the my_all? , my_any? and my_none? to respond when you pass a class as an argument and when you pass a regular expression.

Updated the README so that it's more descriptive.
Couldn't change the name of my pull request since am using the same PR but that is something I shall start doing for future projects

File: main.rb [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+            self.my_each { |item| result.push(item) if yield(item) }
+            result
+        rescue => e
+            puts "Exception Class: #{e.class.name}"
+        end
+    end
+
+    def my_all?(value = nil)
+        begin
+            if value == nil && !block_given?
```
[the remark this answers is not in the document]
Comment (maelfosso) [reply main.rb]:
```ruby
irb(main):013:0> array
=> [0, 8, 0, 6, 4, 4, 3, 5, 7, 2, 8, 7, 4, 6, 2, 3, 0, 0, 4, 8, 1, 7, 8, 2, 8, 5, 5, 8, 4, 0, 6, 7, 4, 3, 1, 8, 5, 5, 6, 3, 3, 0, 0, 0, 5, 8, 0, 1, 3, 2, 4, 5, 4, 4, 0, 5, 0, 0, 5, 2, 7, 6, 0, 5, 4, 7, 2, 2, 7, 6, 8, 6, 7, 5, 0, 6, 7, 4, 6, 5, 4, 7, 1, 6, 0, 3, 0, 7, 8, 3, 6, 0, 7, 2, 1, 5, 7, 4, 4, 0]
irb(main):014:0> array.all?(Integer)
=> true
irb(main):015:0> array.my_all?(Integer)
=> false
irb(main):016:0> 
```

[the remark this answers is not in the document]
Comment (maelfosso) [reply main.rb]:
```ruby
irb(main):018:0> words = %w[dog door rod blade]
=> ["dog", "door", "rod", "blade"]
irb(main):019:0> words.all?(/d/)
=> true
irb(main):020:0> words.my_all?(/d/)
=> false
irb(main):021:0> 
```

File: main.rb [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+              my_each { |item| return true if item.class == value }
+              false
+          elsif value.class == Regexp
+              my_each { |item| return true if item =~ value }
+              false
+          end
+        rescue => e
+            puts "Exception Class: #{e.class.name}"
+      end
+    end
```
Comment (maelfosso) [inline main.rb]:
```ruby
irb(main):023:0> array
=> [0, 8, 0, 6, 4, 4, 3, 5, 7, 2, 8, 7, 4, 6, 2, 3, 0, 0, 4, 8, 1, 7, 8, 2, 8, 5, 5, 8, 4, 0, 6, 7, 4, 3, 1, 8, 5, 5, 6, 3, 3, 0, 0, 0, 5, 8, 0, 1, 3, 2, 4, 5, 4, 4, 0, 5, 0, 0, 5, 2, 7, 6, 0, 5, 4, 7, 2, 2, 7, 6, 8, 6, 7, 5, 0, 6, 7, 4, 6, 5, 4, 7, 1, 6, 0, 3, 0, 7, 8, 3, 6, 0, 7, 2, 1, 5, 7, 4, 4, 0]
irb(main):024:0> array.any?(Integer)
=> true
irb(main):025:0> array.my_any?(Integer)
=> false
irb(main):026:0> 
```

Comment (maelfosso) [reply main.rb]:
```ruby
irb(main):028:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):029:0> words.any?(/d/)
=> true
irb(main):030:0> words.my_any?(/d/)
=> false
irb(main):031:0>
```

Comment (rmauritsson) [reply main.rb]: Modified and tested them they should be working now.

**Comment by rmauritsson** [comment]
[my_any tests](https://drive.google.com/file/d/1eii2movCbF-mCQJl3Z8K0DzUMwVqg3JW/view?usp=sharing)

Modified the method and tested it as shown in the screenshot

**Pull request merged**